apicast/src/apicast.lua

```diff
@@ -65,26 +67,45 @@ function _M:rewrite()
   ngx.ctx.proxy = p
 end
 
-function _M.post_action()
+function _M:post_action()
   local request_id = ngx.var.original_request_id
+  local post_action_proxy = self.post_action_proxy
+
+  if not post_action_proxy then
+    return nil, 'not initialized'
+  end
+
   local p = ngx.ctx.proxy or post_action_proxy[request_id]
 
   post_action_proxy[request_id] = nil
 
   if p then
-    p:post_action()
+    return p:post_action()
   else
     ngx.log(ngx.INFO, 'could not find proxy for request id: ', request_id)
```

Comment:
This info message looks a bit alarming, like if it was supposed to find proxy, but couldn't, while most of the times it is absolutely fine and deliberate. Maybe say "skipping post_action, as no proxy was registered for the request id", and move it to DEBUG level?...